o2a/mappers/git_mapper.py

```diff
@@ -63,52 +65,53 @@ def __init__(
         self,
         oozie_node: Element,
         name: str,
+        props: PropertySet,
         trigger_rule: str = TriggerRule.ALL_SUCCESS,
-        params: Dict[str, str] = None,
         **kwargs,
     ):
-        ActionMapper.__init__(self, oozie_node=oozie_node, name=name, trigger_rule=trigger_rule, **kwargs)
-        if params is None:
-            params = {}
-        self.params = params
-        self.bash_command: Optional[str] = None
+        ActionMapper.__init__(
+            self, oozie_node=oozie_node, name=name, trigger_rule=trigger_rule, props=props, **kwargs
+        )
+        self.git_uri: Optional[str] = None
+        self.git_branch: Optional[str] = None
+        self.destination_uri: Optional[str] = None
+        self.destination_path: Optional[str] = None
+        self.key_path_uri: Optional[str] = None
+        self.key_path: Optional[str] = None
+        self.prepare_extension: PrepareMapperExtension = PrepareMapperExtension(self)
 
     def on_parse_node(self):
-        git_uri = get_tag_el_text(self.oozie_node, TAG_GIT_URI, self.params)
-        git_branch = get_tag_el_text(self.oozie_node, TAG_BRANCH, self.params)
-        destination_uri = get_tag_el_text(self.oozie_node, TAG_DESTINATION_URI, self.params)
-        destination_path = urlparse(destination_uri).path
-        key_path_uri = get_tag_el_text(self.oozie_node, TAG_KEY_PATH, self.params)
-        key_path = urlparse(key_path_uri).path
-        self.bash_command = prepare_git_command(
-            git_uri=git_uri, git_branch=git_branch, destination_path=destination_path, key_path=key_path
+        super().on_parse_node()
+        self.git_uri = get_tag_el_text(self.oozie_node, TAG_GIT_URI, props=self.props, default="")
+        self.git_branch = get_tag_el_text(self.oozie_node, TAG_BRANCH, props=self.props, default="")
+        self.destination_uri = get_tag_el_text(
+            self.oozie_node, tag=TAG_DESTINATION_URI, props=self.props, default=""
+        )
+        self.destination_path = urlparse(self.destination_uri).path
+        self.key_path_uri = get_tag_el_text(self.oozie_node, tag=TAG_KEY_PATH, props=self.props, default="")
+        self.key_path = urlparse(self.key_path_uri).path
+
+    def to_tasks_and_relations(self) -> Tuple[List[Task], List[Relation]]:
+        action_task = Task(
+            task_id=self.name,
+            template_name="git.tpl",
```

Comment:
It looks like there are no tests for this template. I tried to find the change in the tests corresponding to the change and there is no test class

Reply:
Good catch! I will add it as separate case

Reply:
You already added it. Merged